src/Bicep.Core/Emit/ExpressionConverter.cs

```diff
@@ -70,6 +71,13 @@ public LanguageExpression ConvertExpression(SyntaxBase expression)
                         Array.Empty<LanguageExpression>());
 
                 case FunctionCallSyntax function:
+                    var returnValueType = context.SemanticModel.GetTypeInfo(function);
+                    if (returnValueType is IResourceScopeType resourceScopeType && !ScopeHelper.CanConvertToArmJson(resourceScopeType))
```

Comment:
>CanConvertToArmJson [](start = 96, length = 19)

Should we just call the converter and have it return {} as a valid conversion instead. it would remove the duplication below?